### DISCUSSION TITLE
Why do my "recent commits" sometimes disappear and how do I get them back?

### DISCUSSION THREAD

**Discussion by dabrahams** [magit/magit#5063] [Support] [answered]
Title says it all.  Relevant info:

Magit v3.3.0-841-g42e261d2, Transient 0.4.3, Git 2.42.1, Emacs 29.1, darwin

magit-log-section-commit-count is a variable defined in ‘magit-log.el’.

Its value is 10

**Comment by tarsius** [comment]
This section might be inserted by this:

```elisp
(defun magit-insert-unpushed-to-upstream-or-recent ()
  "Insert section showing unpushed or other recent commits.
If an upstream is configured for the current branch and it is
behind of the current branch, then show the commits that have
not yet been pushed into the upstream branch.  If no upstream is
configured or if the upstream is not behind of the current branch,
then show the last `magit-log-section-commit-count' commits."
  (let ((upstream (magit-get-upstream-branch)))
    (if (or (not upstream)
            (magit-rev-ancestor-p "HEAD" upstream))
        (magit-insert-recent-commits 'unpushed "@{upstream}..")
      (magit-insert-unpushed-to-upstream))))
```

You can remove that function from `magit-status-section-hook` and instead add `magit-insert-unpushed-to-upstream` and `magit-insert-recent-commits`. These section will then often list the same commits; but on the other hand, "unpushed to upstream" might show fewer commits than "N most recent", so you might still find this desirable.

**Reply by dabrahams** [reply, marked as answer]
For anyone reading this answer, it's `magit-status-sections-hook`.  This seems to be working for me:

```elisp
  ;; https://github.com/magit/magit/discussions/5063#discussioncomment-7809087
  (let ((h 'magit-status-sections-hook))
    (remove-hook h 'magit-insert-unpushed-to-upstream-or-recent)
    (add-hook h 'magit-insert-unpushed-to-upstream 100)
    (add-hook h 'magit-insert-recent-commits 100))
```

Thanks @tarsius !
